packages/core-blockchain/src/processor/block-processor.ts

```diff
@@ -112,4 +117,44 @@ export class BlockProcessor {
 
         return false;
     }
+
+    /**
+     * Check if a block's transactions violate nonce order.
+     * For a given sender, all v1 transactions must precede all v2 transactions.
+     * In addition, for a given sender, v2 transactions must have strictly
+     * increasing nonce without gaps.
+     */
+    private blockContainsOutOfOrderNonce(block: Interfaces.IBlock): boolean {
+        const nonceBySender = {};
+
+        for (const transaction of block.transactions) {
+            const data = transaction.data;
+
+            if (data.version < 2) {
```

Comment:
I think if a block mixes V1 and V2 transactions it should be rejected.

Reply:
Done in 4b07b8b26.